// react-bulma-components/lib/components/tabs/tabs.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames2 = require('classnames');

var _classnames3 = _interopRequireDefault(_classnames2);

var _tab = require('./components/tab');

var _tab2 = _interopRequireDefault(_tab);

var _modifiers = require('../../modifiers');

var _modifiers2 = _interopRequireDefault(_modifiers);

var _element = require('../element');

var _element2 = _interopRequireDefault(_element);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var Tabs = function Tabs(_ref) {
  var _classnames;

  var children = _ref.children,
      className = _ref.className,
      align = _ref.align,
      size = _ref.size,
      type = _ref.type,
      fullwidth = _ref.fullwidth,
      props = _objectWithoutProperties(_ref, ['children', 'className', 'align', 'size', 'type', 'fullwidth']);

  return _react2.default.createElement(
    _element2.default,
    _extends({}, props, {
      className: (0, _classnames3.default)('tabs', className, (_classnames = {}, _defineProperty(_classnames, 'is-' + align, align), _defineProperty(_classnames, 'is-' + size, size), _defineProperty(_classnames, 'is-toggle', type === 'toggle-rounded'), _defineProperty(_classnames, 'is-' + type, type), _defineProperty(_classnames, 'is-fullwidth', fullwidth), _classnames))
    }),
    _react2.default.createElement(
      'ul',
      null,
      children
    )
  );
};

Tabs.Tab = _tab2.default;

Tabs.propTypes = _extends({}, _modifiers2.default.propTypes, {
  children: _propTypes2.default.node,
  className: _propTypes2.default.string,
  style: _propTypes2.default.shape({}),
  renderAs: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.func]),
  align: _propTypes2.default.oneOf(['centered', 'right']),
  size: _propTypes2.default.oneOf(['small', 'medium', 'large']),
  /** This is called style on Bulma documentation */
  type: _propTypes2.default.oneOf(['toggle', 'boxed', 'toggle-rounded']),
  fullwidth: _propTypes2.default.bool
});

Tabs.defaultProps = _extends({}, _modifiers2.default.defaultProps, {
  children: null,
  className: '',
  style: {},
  renderAs: 'div',
  align: null,
  size: null,
  type: null,
  fullwidth: false
});

exports.default = Tabs;
