// react-bulma-components/lib/components/form/components/select.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames2 = require('classnames');

var _classnames3 = _interopRequireDefault(_classnames2);

var _modifiers = require('../../../modifiers');

var _modifiers2 = _interopRequireDefault(_modifiers);

var _constants = require('../../../constants');

var _constants2 = _interopRequireDefault(_constants);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var colors = [null].concat(Object.keys(_constants2.default.COLORS).map(function (key) {
  return _constants2.default.COLORS[key];
}));

var Select = function Select(_ref) {
  var _classnames;

  var className = _ref.className,
      style = _ref.style,
      size = _ref.size,
      color = _ref.color,
      loading = _ref.loading,
      readOnly = _ref.readOnly,
      disabled = _ref.disabled,
      value = _ref.value,
      multiple = _ref.multiple,
      children = _ref.children,
      name = _ref.name,
      allProps = _objectWithoutProperties(_ref, ['className', 'style', 'size', 'color', 'loading', 'readOnly', 'disabled', 'value', 'multiple', 'children', 'name']);

  var props = _modifiers2.default.clean(allProps);
  return _react2.default.createElement(
    'div',
    {
      className: (0, _classnames3.default)('select', _modifiers2.default.classnames(allProps), className, (_classnames = {}, _defineProperty(_classnames, 'is-' + size, size), _defineProperty(_classnames, 'is-' + color, color), _defineProperty(_classnames, 'is-loading', loading), _defineProperty(_classnames, 'is-multiple', multiple), _classnames)),
      style: style
    },
    _react2.default.createElement(
      'select',
      _extends({}, props, {
        multiple: multiple,
        value: value,
        readOnly: readOnly,
        disabled: disabled,
        name: name
      }),
      children
    )
  );
};

Select.propTypes = _extends({}, _modifiers2.default.propTypes, {
  children: _propTypes2.default.node,
  className: _propTypes2.default.string,
  style: _propTypes2.default.shape({}),
  size: _propTypes2.default.oneOf(['small', 'medium', 'large']),
  color: _propTypes2.default.oneOf(colors),
  readOnly: _propTypes2.default.bool,
  disabled: _propTypes2.default.bool,
  multiple: _propTypes2.default.bool,
  loading: _propTypes2.default.bool,
  value: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.number]),
  /**
       * The name of the input field Commonly used for [multi-input handling](https://reactjs.org/docs/forms.html#handling-multiple-inputs)
       */
  name: _propTypes2.default.string
});

Select.defaultProps = _extends({}, _modifiers2.default.defaultProps, {
  children: null,
  className: '',
  value: '',
  style: {},
  size: null,
  color: null,
  readOnly: false,
  disabled: false,
  multiple: false,
  loading: false,
  name: null
});

exports.default = Select;
