// react-bulma-components/lib/components/breadcrumb/breadcrumb.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames2 = require('classnames');

var _classnames3 = _interopRequireDefault(_classnames2);

var _modifiers = require('../../modifiers');

var _modifiers2 = _interopRequireDefault(_modifiers);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var Breadcrumb = function Breadcrumb(_ref) {
  var _classnames;

  var className = _ref.className,
      items = _ref.items,
      renderAs = _ref.renderAs,
      hrefAttr = _ref.hrefAttr,
      separator = _ref.separator,
      size = _ref.size,
      align = _ref.align,
      allProps = _objectWithoutProperties(_ref, ['className', 'items', 'renderAs', 'hrefAttr', 'separator', 'size', 'align']);

  var Element = renderAs;
  var props = _modifiers2.default.clean(allProps);
  return _react2.default.createElement(
    'nav',
    _extends({}, props, {
      className: (0, _classnames3.default)('breadcrumb', className, _modifiers2.default.classnames(allProps), (_classnames = {}, _defineProperty(_classnames, 'has-' + separator + '-separator', separator), _defineProperty(_classnames, 'is-' + size, size), _defineProperty(_classnames, 'is-' + align, align), _classnames))
    }),
    _react2.default.createElement(
      'ul',
      null,
      items.map(function (item) {
        var p = _defineProperty({}, renderAs === 'a' ? 'href' : hrefAttr, item.url);
        return _react2.default.createElement(
          'li',
          {
            key: item.url,
            className: (0, _classnames3.default)({
              'is-active': item.active
            })
          },
          _react2.default.createElement(
            Element,
            p,
            item.name
          )
        );
      })
    )
  );
};

Breadcrumb.propTypes = _extends({}, _modifiers2.default.propTypes, {
  className: _propTypes2.default.string,
  style: _propTypes2.default.shape({}),
  separator: _propTypes2.default.oneOf(['arrow', 'bullet', 'dot', 'succeeds']),
  size: _propTypes2.default.oneOf(['small', 'medium', 'large']),
  align: _propTypes2.default.oneOf(['right', 'center']),
  items: _propTypes2.default.arrayOf(_propTypes2.default.shape({
    url: _propTypes2.default.string.isRequired,
    active: _propTypes2.default.bool,
    name: _propTypes2.default.node
  })),
  renderAs: _propTypes2.default.oneOfType([_propTypes2.default.oneOf(['a']), _propTypes2.default.func]),
  hrefAttr: _propTypes2.default.string
});

Breadcrumb.defaultProps = _extends({}, _modifiers2.default.defaultProps, {
  items: [],
  hrefAttr: null,
  separator: null,
  renderAs: 'a',
  className: '',
  style: {},
  size: null,
  align: null
});

exports.default = Breadcrumb;
