// react-bulma-components/lib/components/hero/hero.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames2 = require('classnames');

var _classnames3 = _interopRequireDefault(_classnames2);

var _constants = require('../../constants');

var _constants2 = _interopRequireDefault(_constants);

var _heroHead = require('./components/hero-head');

var _heroHead2 = _interopRequireDefault(_heroHead);

var _heroBody = require('./components/hero-body');

var _heroBody2 = _interopRequireDefault(_heroBody);

var _heroFooter = require('./components/hero-footer');

var _heroFooter2 = _interopRequireDefault(_heroFooter);

var _modifiers = require('../../modifiers');

var _modifiers2 = _interopRequireDefault(_modifiers);

var _element = require('../element');

var _element2 = _interopRequireDefault(_element);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var colors = [null].concat(Object.keys(_constants2.default.COLORS).map(function (key) {
  return _constants2.default.COLORS[key];
}));

var Hero = function Hero(_ref) {
  var _classnames;

  var children = _ref.children,
      className = _ref.className,
      color = _ref.color,
      gradient = _ref.gradient,
      size = _ref.size,
      props = _objectWithoutProperties(_ref, ['children', 'className', 'color', 'gradient', 'size']);

  return _react2.default.createElement(
    _element2.default,
    _extends({}, props, {
      className: (0, _classnames3.default)('hero', className, (_classnames = {}, _defineProperty(_classnames, 'is-' + color, color), _defineProperty(_classnames, 'is-' + size, size), _defineProperty(_classnames, 'is-bold', gradient), _classnames))
    }),
    children
  );
};

Hero.Head = _heroHead2.default;

Hero.Body = _heroBody2.default;

Hero.Footer = _heroFooter2.default;

Hero.propTypes = _extends({}, _modifiers2.default.propTypes, {
  children: _propTypes2.default.node,
  className: _propTypes2.default.string,
  style: _propTypes2.default.shape({}),
  renderAs: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.func]),
  color: _propTypes2.default.oneOf(colors),
  gradient: _propTypes2.default.bool,
  size: _propTypes2.default.oneOf(['medium', 'large', 'fullheight'])
});

Hero.defaultProps = _extends({}, _modifiers2.default.defaultProps, {
  children: null,
  className: '',
  style: {},
  renderAs: 'section',
  color: null,
  gradient: false,
  size: null
});

exports.default = Hero;
