// react-bulma-components/lib/components/progress/progress.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames2 = require('classnames');

var _classnames3 = _interopRequireDefault(_classnames2);

var _constants = require('../../constants');

var _constants2 = _interopRequireDefault(_constants);

var _modifiers = require('../../modifiers');

var _modifiers2 = _interopRequireDefault(_modifiers);

var _element = require('../element');

var _element2 = _interopRequireDefault(_element);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var colors = [null].concat(Object.keys(_constants2.default.COLORS).map(function (key) {
  return _constants2.default.COLORS[key];
}));

var Progress = function Progress(_ref) {
  var _classnames;

  var className = _ref.className,
      value = _ref.value,
      max = _ref.max,
      color = _ref.color,
      size = _ref.size,
      props = _objectWithoutProperties(_ref, ['className', 'value', 'max', 'color', 'size']);

  return _react2.default.createElement(_element2.default, _extends({
    renderAs: 'progress'
  }, props, {
    value: value,
    max: max,
    className: (0, _classnames3.default)('progress', className, (_classnames = {}, _defineProperty(_classnames, 'is-' + color, color), _defineProperty(_classnames, 'is-' + size, size), _classnames))
  }));
};

Progress.propTypes = _extends({}, _modifiers2.default.propTypes, {
  className: _propTypes2.default.string,
  style: _propTypes2.default.shape({}),
  color: _propTypes2.default.oneOf(colors),
  size: _propTypes2.default.oneOf(['small', 'medium', 'large']),
  value: _propTypes2.default.number.isRequired,
  max: _propTypes2.default.number.isRequired
});

Progress.defaultProps = _extends({}, _modifiers2.default.defaultProps, {
  className: '',
  style: {},
  color: null,
  size: null
});

exports.default = Progress;
