// react-bulma-components/lib/modifiers/responsives.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames2 = require('classnames');

var _classnames3 = _interopRequireDefault(_classnames2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var displays = ['block', 'flex', 'inline', 'inline-block', 'inline-flex'];

var getSizeClassFromProp = function getSizeClassFromProp(sizes) {
  return Object.keys(sizes).reduce(function (classes, size) {
    var _extends2;

    var display = sizes[size].display || {};
    var hide = sizes[size].hide || {};
    var textSize = sizes[size].textSize || {};
    var textAlignment = sizes[size].textAlignment || {};

    var obj = _extends({}, classes, (_extends2 = {}, _defineProperty(_extends2, 'is-' + display.value + '-' + size + (display.only ? '-only' : ''), display.value), _defineProperty(_extends2, 'is-hidden-' + size + (hide.only ? '-only' : ''), hide.value), _defineProperty(_extends2, 'has-text-' + textAlignment.value + '-' + size + (textAlignment.only ? '-only' : ''), textAlignment.value), _defineProperty(_extends2, 'is-size-' + textSize.value + '-' + size, textSize.value > 0), _extends2));

    return obj;
  }, {});
};

var sizeShape = _propTypes2.default.shape({
  display: _propTypes2.default.shape({
    value: _propTypes2.default.oneOf(displays),
    only: _propTypes2.default.bool
  }),
  hide: _propTypes2.default.shape({
    value: _propTypes2.default.bool,
    only: _propTypes2.default.bool
  }),
  textSize: _propTypes2.default.shape({
    value: _propTypes2.default.oneOf([1, 2, 3, 4, 5, 6])
  }),
  textAlignment: _propTypes2.default.shape({
    value: _propTypes2.default.oneOf(['centered', 'justified', 'left', 'right']),
    only: _propTypes2.default.bool
  })
});

exports.default = {
  propTypes: {
    responsive: _propTypes2.default.shape({
      mobile: sizeShape,
      tablet: sizeShape,
      desktop: sizeShape,
      widescreen: sizeShape,
      fullhd: sizeShape,
      touch: sizeShape
    })
  },
  defaultProps: {
    responsive: undefined
  },
  classnames: function classnames(props) {
    return (0, _classnames3.default)(_extends({}, getSizeClassFromProp(props.responsive || {})));
  },
  clean: function clean(_ref) {
    var responsive = _ref.responsive,
        hide = _ref.hide,
        props = _objectWithoutProperties(_ref, ['responsive', 'hide']);

    return props;
  }
};
