// react-bulma-components/lib/components/navbar/components/burger.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _context = require('../context');

var _modifiers = require('../../../modifiers');

var _modifiers2 = _interopRequireDefault(_modifiers);

var _element = require('../../element');

var _element2 = _interopRequireDefault(_element);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var NavbarBurger = function NavbarBurger(_ref) {
  var style = _ref.style,
      className = _ref.className,
      allProps = _objectWithoutProperties(_ref, ['style', 'className']);

  var props = _modifiers2.default.clean(allProps);
  return _react2.default.createElement(
    _context.ShowContext.Consumer,
    null,
    function (active) {
      return _react2.default.createElement(
        _element2.default,
        _extends({
          role: 'button',
          tabIndex: '0',
          style: _extends({ outline: 'none' }, style),
          className: (0, _classnames2.default)('navbar-burger', _modifiers2.default.classnames(allProps), className, {
            'is-active': active
          })
        }, props),
        _react2.default.createElement('span', null),
        _react2.default.createElement('span', null),
        _react2.default.createElement('span', null)
      );
    }
  );
};

NavbarBurger.propTypes = _extends({}, _modifiers2.default.propTypes, {
  style: _propTypes2.default.shape({}),
  className: _propTypes2.default.string,
  onClick: _propTypes2.default.func
});

NavbarBurger.defaultProps = _extends({}, _modifiers2.default.defaultProps, {
  style: {},
  className: '',
  onClick: function onClick() {}
});

exports.default = NavbarBurger;
