// react-bulma-components/lib/components/navbar/components/container.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames2 = require('classnames');

var _classnames3 = _interopRequireDefault(_classnames2);

var _modifiers = require('../../../modifiers');

var _modifiers2 = _interopRequireDefault(_modifiers);

var _element = require('../../element');

var _element2 = _interopRequireDefault(_element);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var NavbarContainer = function NavbarContainer(_ref) {
  var className = _ref.className,
      children = _ref.children,
      position = _ref.position,
      props = _objectWithoutProperties(_ref, ['className', 'children', 'position']);

  return _react2.default.createElement(
    _element2.default,
    _extends({}, props, {
      className: (0, _classnames3.default)(_defineProperty({}, 'navbar-' + position, position), className)
    }),
    children
  );
};

NavbarContainer.propTypes = _extends({}, _modifiers2.default.propTypes, {
  style: _propTypes2.default.shape({}),
  className: _propTypes2.default.string,
  children: _propTypes2.default.node,
  renderAs: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.func]),
  position: _propTypes2.default.oneOf(['start', 'end'])
});

NavbarContainer.defaultProps = _extends({}, _modifiers2.default.defaultProps, {
  style: {},
  className: '',
  children: null,
  renderAs: 'div',
  position: 'start'
});

exports.default = NavbarContainer;
