// react-bulma-components/lib/components/navbar/components/dropdown.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _modifiers = require('../../../modifiers');

var _modifiers2 = _interopRequireDefault(_modifiers);

var _element = require('../../element');

var _element2 = _interopRequireDefault(_element);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var NavbarDropdown = function NavbarDropdown(_ref) {
  var className = _ref.className,
      boxed = _ref.boxed,
      right = _ref.right,
      children = _ref.children,
      props = _objectWithoutProperties(_ref, ['className', 'boxed', 'right', 'children']);

  return _react2.default.createElement(
    _element2.default,
    _extends({}, props, {
      className: (0, _classnames2.default)('navbar-dropdown', className, {
        'is-boxed': boxed,
        'is-right': right
      })
    }),
    children
  );
};

NavbarDropdown.propTypes = _extends({}, _modifiers2.default.propTypes, {
  style: _propTypes2.default.shape({}),
  className: _propTypes2.default.string,
  children: _propTypes2.default.node,
  renderAs: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.func]),
  boxed: _propTypes2.default.bool,
  right: _propTypes2.default.bool
});

NavbarDropdown.defaultProps = _extends({}, _modifiers2.default.defaultProps, {
  style: {},
  className: '',
  children: null,
  renderAs: 'span',
  boxed: false,
  right: false
});

exports.default = NavbarDropdown;
