// react-bulma-components/lib/components/dropdown/components/item.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _modifiers = require('../../../modifiers');

var _modifiers2 = _interopRequireDefault(_modifiers);

var _element = require('../../element');

var _element2 = _interopRequireDefault(_element);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var DropdownItem = function DropdownItem(_ref) {
  var active = _ref.active,
      children = _ref.children,
      value = _ref.value,
      props = _objectWithoutProperties(_ref, ['active', 'children', 'value']);

  return _react2.default.createElement(
    _element2.default,
    _extends({
      title: value
    }, props, {
      role: 'presentation',
      className: (0, _classnames2.default)('dropdown-item', {
        'is-active': active
      })
    }),
    children
  );
};

DropdownItem.propTypes = _extends({}, _modifiers2.default.propTypes, {
  active: _propTypes2.default.bool,
  children: _propTypes2.default.node,
  value: _propTypes2.default.any.isRequired,
  onClick: _propTypes2.default.func
});

DropdownItem.defaultProps = _extends({}, _modifiers2.default.defaultProps, {
  active: false,
  onClick: undefined,
  children: null
});

exports.default = DropdownItem;
