// react-bulma-components/lib/components/modal/components/card/head.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _button = require('../../../button');

var _button2 = _interopRequireDefault(_button);

var _modifiers = require('../../../../modifiers');

var _modifiers2 = _interopRequireDefault(_modifiers);

var _element = require('../../../element');

var _element2 = _interopRequireDefault(_element);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var ModalCardHead = function ModalCardHead(_ref) {
  var children = _ref.children,
      className = _ref.className,
      showClose = _ref.showClose,
      onClose = _ref.onClose,
      props = _objectWithoutProperties(_ref, ['children', 'className', 'showClose', 'onClose']);

  return _react2.default.createElement(
    _element2.default,
    _extends({}, props, {
      className: (0, _classnames2.default)('modal-card-head', className)
    }),
    children,
    showClose && _react2.default.createElement(_button2.default, { remove: true, onClick: onClose })
  );
};

ModalCardHead.propTypes = _extends({}, _modifiers2.default.propTypes, {
  children: _propTypes2.default.node,
  className: _propTypes2.default.string,
  style: _propTypes2.default.shape({}),
  showClose: _propTypes2.default.bool,
  onClose: _propTypes2.default.func
});

ModalCardHead.defaultProps = _extends({}, _modifiers2.default.defaultProps, {
  children: null,
  className: '',
  style: {},
  showClose: true,
  onClose: null,
  renderAs: 'header'
});

exports.default = ModalCardHead;
