// react-bulma-components/lib/components/form/components/field/field.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames2 = require('classnames');

var _classnames3 = _interopRequireDefault(_classnames2);

var _fieldLabel = require('./field-label');

var _fieldLabel2 = _interopRequireDefault(_fieldLabel);

var _fieldBody = require('./field-body');

var _fieldBody2 = _interopRequireDefault(_fieldBody);

var _modifiers = require('../../../../modifiers');

var _modifiers2 = _interopRequireDefault(_modifiers);

var _element = require('../../../element');

var _element2 = _interopRequireDefault(_element);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var Field = function Field(_ref) {
  var _classnames;

  var className = _ref.className,
      align = _ref.align,
      multiline = _ref.multiline,
      horizontal = _ref.horizontal,
      kind = _ref.kind,
      props = _objectWithoutProperties(_ref, ['className', 'align', 'multiline', 'horizontal', 'kind']);

  var k = null;

  if (kind === 'addons') {
    k = 'has-addons';
  } else if (kind === 'group') {
    k = 'is-grouped';
  }

  return _react2.default.createElement(_element2.default, _extends({}, props, {
    className: (0, _classnames3.default)('field', className, (_classnames = {}, _defineProperty(_classnames, '' + k, k), _defineProperty(_classnames, k + '-' + align, k && align), _defineProperty(_classnames, k + '-multiline', k === 'is-grouped' && multiline), _defineProperty(_classnames, 'is-horizontal', horizontal), _classnames))
  }));
};

Field.Label = _fieldLabel2.default;

Field.Body = _fieldBody2.default;

Field.propTypes = _extends({}, _modifiers2.default.propTypes, {
  children: _propTypes2.default.node,
  className: _propTypes2.default.string,
  style: _propTypes2.default.shape({}),
  renderAs: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.func]),
  align: _propTypes2.default.oneOf(['centered', 'right']),
  kind: _propTypes2.default.oneOf(['addons', 'group']),
  multiline: _propTypes2.default.bool,
  horizontal: _propTypes2.default.bool
});

Field.defaultProps = _extends({}, _modifiers2.default.defaultProps, {
  children: null,
  className: '',
  style: {},
  renderAs: 'div',
  align: null,
  kind: null,
  multiline: false,
  horizontal: false
});

exports.default = Field;
